ci/verify-build.sh

```diff
@@ -237,7 +239,15 @@ for target in $targets; do
             TARGET="$target" ./ci/install-rust.sh
             test_target "$target"
         else
-            test_target "$target"
+            if [ "$rust" = "stable" ] || [ "$rust" = "nightly" ]; then
+                test_target "$target"
+            else
+                if [ "$target" = "wasm32-wasip1" ]; then
+                    test_target "wasm32-wasi"
+                elif [ "$target" != "wasm32-wasip2" ]; then
+                    test_target "$target"
+                fi
+            fi
```

Comment:
I think this might be cleaner as a case statement

```sh
case "$target" in
    "stable") use_wasi_pn=1 ;;
    "beta") use_wasi_pn=1 ;;
    "nightly") use_wasi_pn=1 ;;
    *) use_wasi_pn=0 ;;
esac

# `wasm32-wasip1` was renamed from `wasm32-wasi`
if [ "$target" = "wasm32-wasip1" ] && [ "$use_wasi_pn" = "0" ]; then
    target="wasm32-wasi"
fi

# `wasm32-wasip2` only exists in recent versions of Rust
if [ "$target" = "wasm32-wasip2" ] && [ "$use_wasi_pn" = "0" ]; then
    continue
fi

test_target "$target"
```

Reply:
Done